src/client/src/.tests/components/chatpage.test.js

```diff
@@ -0,0 +1,164 @@
+/* eslint-enable no-unused-vars */
+let suite = {}
+
+global.window = {}
+
+configure({adapter: new Adapter()})
+describe('<ChatPage />', () => {
+	beforeEach(() => {
+		suite.wrapper = mount(<BrowserRouter><ChatPage location={{state: {username: DUMMY_USER}}}/></BrowserRouter>)
+		suite.Component = shallow(<BrowserRouter><ChatPage location={{state: {username: DUMMY_USER}}}/></BrowserRouter>).find(ChatPage).dive()
+		window.sessionStorage = global.sessionStorage
+	})
+	afterEach(() => {
+		suite = {}
+		window.sessionStorage.clear()
+
+	})
+	it('Should render without crashing', () => {
+		expect(suite.wrapper.length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Should have a ConnectStatus component', () => {
+		expect(suite.wrapper.find(ConnectStatus).length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Should have a sidebar', () => {
+		expect(suite.wrapper.find(SIDEBAR).length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Should have no rooms loaded', () => {
+		expect(suite.wrapper.find(Room).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should have a message-area', () => {
+		expect(suite.wrapper.find(MESSAGE_AREA).length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Should have no messages loaded', () => {
+		expect(suite.wrapper.find(MESSAGE_CLASS).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should have a logout button', () => {
+		expect(suite.wrapper.find('a').length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Logout button URL should be / ', () => {
+		expect(suite.wrapper.find({href: HOME_PATH}).length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Should return change the selected room', () => {
+ 
+		suite.Component.instance().changeSelectedRoom({roomId: DUMMY_ROOM})
+		expect(suite.Component.state(SELECTED_ROOM)).toBe(DUMMY_ROOM)
+	})
+	it('Should have no selected room on mount', () => {
+		expect(suite.Component.state(SELECTED_ROOM)).toBe(EMPTY_MAGIC_STRING)
+	})
+	it('Should be disconnected at startup', () => {
+		expect(suite.Component.state(CONNECTED)).toBe(FALSE)
+	})
+	it('Should have no input at startup', () => {
+		expect(suite.Component.state(INPUT)).toBe(EMPTY_MAGIC_STRING)
+	})
+	it('Should have no messages at startup', () => {
+		expect(suite.Component.state(MESSAGES).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should have no userRooms at startup', () => {
+		expect(suite.Component.state(USERROOMS).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should have a username at startup', () => {
+		expect(suite.Component.state(USERNAME)).toBe(DUMMY_USER)
+	})
+	it('Should update joined rooms', () => {
+		const fakeData = {roomId: DUMMY_ROOM, usernames: DUMMY_USER}
+		suite.Component.instance().updateJoinedRooms(fakeData)
+		expect(suite.Component.state(USERROOMS)[0].roomId).toBe(fakeData.roomId)
+		expect(suite.Component.state(USERROOMS)[0].usernames).toBe(fakeData.usernames)
+	})
+	it('Should store a message to sessionStorage', () => {
+		const roomId = DUMMY_ROOM
+		const fakeData = {username: DUMMY_USER, message: DUMMY_MESSAGE, timestamp: new Date().getTime()}
```

Comment:
A good rule is to not use random things in tests, unless it's the precise task to check in test. Mock timestamp instead of generating it.
Adjust other tests